src/test/java/SuiteClassIntercepter.java

```diff
@@ -0,0 +1,71 @@
+import com.browserstack.local.Local;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.testng.ISuite;
+import org.testng.ISuiteListener;
+
+import java.io.FileReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class SuiteClassIntercepter implements ISuiteListener {
+    Local local;
+
+    @Override
+    public void onStart(ISuite iSuite) {
+        String profile = iSuite.getParameter("profile");
+        try {
+            JSONParser parser = new JSONParser();
+            JSONObject config = (JSONObject) parser.parse(new FileReader("config/browserstack.conf.json"));
+            Map<String, Map<String, String>> profiles = (Map<String, Map<String, String>>) (config.get("profiles"));
+            if (profile != null) {
+                Map<String, String> profileMap = profiles.get(profile);
+                if (profileMap.get("browserstack.local") != null
+                        && Objects.equals(profileMap.get("browserstack.local"), "true")) {
+                    System.out.println("Starting BrowserStack Local...");
+                    local = new Local();
```

Comment:
How are we passing localOptions? There should be a way where we can read from JSON and pass it while starting Local